src/ngx_http_lua_regex.c

```diff
@@ -1116,12 +1116,12 @@ ngx_http_lua_ngx_re_gmatch_iterator(lua_State *L)
 
     dd("rc = %d", (int) rc);
 
-    lua_createtable(L, rc /* narr */, 0 /* nrec */);
+    lua_createtable(L, ctx->ncaptures + 1 /* narr */, name_count /* nrec */);
```

Comment:
Okay, in case that `ctx->ncaptures` is 0, we indeed need to plus one otherwise LuaJIT will resize (auto-grow) the table, which is expensive (for narr > 0, LuaJIT automatically adds one to the user-supplied value since LuaJIT stores the index 0 in the array part of tables too).